app/full_text_search.py

```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
+    Returns:
+        fullname fields, romanized_name_list: (for check)
+    """
+    fields = []
+    romanized_name_list = []
+    romanized_given_name = romanize_method(given_name)
+    romanized_family_name = romanize_method(family_name)
+    full_name_given_family = create_full_name_without_space(
+        romanized_given_name, romanized_family_name
+    )
+    full_name_family_given = create_full_name_without_space(
+        romanized_family_name, romanized_given_name
+    )
+    if full_name_given_family:
+        fields.append(appengine_search.TextField(
+            name='no_spacefull_name_1_'+romanize_method.__name__,
+            value=full_name_given_family))
+        romanized_name_list.append(full_name_given_family)
+    if full_name_family_given:
+        fields.append(appengine_search.TextField(
+            name='no_spacefull_name_2_'+romanize_method.__name__,
+            value=full_name_family_given))
+        romanized_name_list.append(full_name_family_given)
+    return fields, romanized_name_list
+
+
+def create_romanized_name_fields(romanize_method, **kwargs):
+    """
+    Creates fields(romanized by romanize_method) for full text search.
     """
     fields = []
     romanized_names_list = []
+    romanize_method_name = romanize_method.__name__
     for field in kwargs:
-        if kwargs[field] and (re.match(ur'([\u3400-\u9fff])', kwargs[field])):
-            romanized_japanese_name = (
-                script_variant.romanize_japanese_name_by_name_dict(
-                    kwargs[field]))
-            if romanized_japanese_name:
-                fields.append(
-                    appengine_search.TextField(
-                        name=field+'_romanized_by_jp_name_dict',
-                        value=romanized_japanese_name)
-                )
-                romanized_names_list.append(romanized_japanese_name)
-            
-    # field for checking if query words contian a part of person name.
-    romanized_jp_names = (
-            ':'.join([name for name in romanized_names_list if name]))
-    fields.append(appengine_search.TextField(name='romanized_jp_names',
-                                             value=romanized_jp_names))
+        romanized_name = romanize_method(kwargs[field])
+        if romanized_name:
+            fields.append(
+                appengine_search.TextField(
+                    name=field+'_romanized_by_'+romanize_method_name,
+                    value=romanized_name))
+            romanized_names_list.append(romanized_name)
+
+    full_name_field, romanized_full_names = create_full_name_without_space_fields(
+        romanize_method, kwargs['given_name'], kwargs['family_name'])
+    fields.extend(full_name_field)
+    romanized_names_list.extend(romanized_full_names)
+
+    names = ':'.join([name for name in romanized_names_list if name])
+    fields.append(
+        appengine_search.TextField(name='name_'+romanize_method_name,
```

Comment:
Line 173 generates {field}_romanized_by_{method_name}, but here generates name_{method_name}. Can you be consistent with the naming?

Also, the field name here should be "names", not "name".